src/test/perl/path.t

```diff
@@ -108,9 +108,9 @@ sub verify_exception
     } else {
         ok(! $ec_check->error(), "Error reset after $msg");
     };
-    if ($noreset) {
+    if ($noreset && $mc->{fail}) {
```

Comment:
nice catch on the `res_txt`! (i'm waiting for https://github.com/quattor/maven-tools/issues/130 to catch these sort of errors)

if you want to, you can make the tests more robust, but not by changing the logic like this.